instaloader/instaloader.py

```diff
@@ -204,6 +206,33 @@ def __init__(self,
         self.resume_prefix = resume_prefix
         self.check_resume_bbd = check_resume_bbd
 
+        self.slide = slide or ""
+        self.slide_start = 0
+        self.slide_end = 0
```

Comment:
If I understand this if-nesting right, we have the following cases:

1. no `--slide` parameter: *start=1, end=0*, however *start=0, end=0* is the default for get_sidecar_posts(),
1. `--slide NUMBER`: *start=NUMBER, end=NUMBER*,
1. `--slide last`: *start=0, end=-1*,
1. `--slide A-B`, *A* and *B* both numbers: *start=A, end=B*,
1. `--slide NUMBER-last`: *start=NUMBER, end=-1*.

I find the meaning of 0 extremely confusing, i.e. the consistency especially regarding (1) and (3). I also suspect here, that 0 has a different meaning for *start* than for *end*. Isn't there a simpler and less confusing way?

Reply:
(with my suggestion below at the get_sidecar_posts() comment, default (case (1)) becomes *start=0, end=-1*, and we would translate the string input like: *number -> number-1*, *last -> -1*. Case (3) would become *start=-1, end=-1* and as such just like (2) with *-1* instead of *NUMBER-1*)

Reply:
You are right @aandergr, we can definitely think on a less confusing way to use 0 and -1.